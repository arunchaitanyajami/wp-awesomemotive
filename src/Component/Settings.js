import {
	__experimentalInputControl as InputControl,
	ToggleControl,
} from '@wordpress/components';
import { useState, useEffect } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import * as defaultOptions from '../deafultOptions';
import { Icon, plus, close } from '@wordpress/icons';
import EmailInputComponent from './EmailInputComponent';

export default ( {
	settings,
	updateEmails,
	updateRows,
	updateDateReadable,
	updateDataLoadFlag,
	updateDataUpdatedFlag
} ) => {
	const { data } = settings;
	const [ value, setValue ] = useState( data.num_rows );
	const [ date, setDate ] = useState( data.human_date );
	const [ emails, setEmails ] = useState( {} );
	const [ reloadComponent, setReloadComponent ] = useState( true );
	const [ isError, setIsError ] = useState( false );
	const [ message, setMessage ] = useState( '' );

	/**
	 * Set no of rows to display.
	 *
	 * @param value
	 */
	const setRows = ( value ) => {
		setValue( value );
		updateRows( value );
		setReloadComponent(false)
	};

	/**
	 * Updated human readable flag.
	 *
	 * @param booleanValue
	 */
	const setDateReadable = ( booleanValue ) => {
		setDate( booleanValue );
		updateDateReadable( booleanValue );
		setReloadComponent(false)
	};

	/**
	 * Set email count.
	 */
	const setNewInput = () => {
		let index = Object.keys( emails ).length;
		setEmails( {
			...emails,
			...{
				[ index ]: '',
			},
		} );
	};

	/**
	 * Fetch values form email child components and append/replace data.
	 * @param key
	 * @param value
	 */
	const sendEmail = ( key, value ) => {
		let appendEmail = {
			[ key ]: value,
		};

		setEmails( {
			...emails,
			...appendEmail,
		} );

		updateEmails( {
			...emails,
			...appendEmail,
		} );

		setReloadComponent(false)
	};

	/**
	 * Remove email from emails object.
	 *
	 * @param key
	 */
	const removeEmail = ( key ) => {
		delete emails[ key ];
		setEmails( { ...emails } );
		updateEmails( { ...emails } );
		setReloadComponent(false)
	};

	const updatedFlags = () => {
		setIsError(false);
		setMessage('')
		updateDataLoadFlag(false);
		updateDataUpdatedFlag(false);
	}

	/**
	 * Updated flags on closing notice.
	 */
	const onCloseNotice = () => {
		setIsError(false);
		setMessage('')
		updateDataLoadFlag(false);
		updateDataUpdatedFlag(false);
	}

	useEffect(() => {
		if (reloadComponent) {
			setEmails(data.emails);
			setValue(data.num_rows);
			setDate(data.human_date ?? false);
		}

		if ((settings.isDataUpdated || settings.isInitialLoad) && 200 !== settings.status) {
			setIsError(true);
			setMessage(settings.message);
		}

		if ((settings.isDataUpdated || settings.isInitialLoad) && 200 === settings.status) {
			setIsError(false);
			setMessage(settings.message);
		}
	}, [data]);

	/**
	 * Provide error notice.
	 *
	 * @returns {JSX.Element}
	 * @constructor
	 */
	const ErrorNotice = ({type}) => (
		<div className={'components-notice is-' + type}>
			<div className={'components-notice-message'}>
				{'error' === type && <p> An error occurred: <code>{message}</code>.</p>}
				{'success' === type && <p>{message}</p>}
			</div>
			<div className={'components-notice-button'} onClick={ () => { onCloseNotice() }} >
				<p><Icon icon={close}/></p>
			</div>
		</div>
	);

	return (
		<div className={'awesomemotive-settings-container-fluid'}>
			{ isError && message && <ErrorNotice type={'error'} /> }
			{ !isError && message && <ErrorNotice type={'success'} /> }
			<div className={ 'awesomemotive-settings-container' }>
				<div className={ 'awesomemotive-settings-form-inputs' }>
					<InputControl
						label={ __( 'No of rows', defaultOptions.text_domain ) }
						value={ value }
						type={ 'number' }
						id={ 'awesomemotive-settings-form-input-no-of-rows' }
						onChange={ ( nextValue ) => setRows( nextValue ?? '' ) }
					/>
				</div>
				<div className={ 'awesomemotive-settings-form-inputs' }>
					<ToggleControl
						label={ __(
							'Human Readable Date',
							defaultOptions.text_domain
						) }
						help={
							date
								? __(
										'Readable date enabled.',
										defaultOptions.text_domain
								  )
								: __(
										'Timestamp displayed',
										defaultOptions.text_domain
								  )
						}
						checked={ date }
						id={ 'awesomemotive-settings-form-input-date' }
						onChange={ ( value ) => {
							setDateReadable( value );
						} }
					/>
				</div>
				<div className={ 'awesomemotive-settings-form-inputs' }>
					<h2>{ __( 'Email List:', defaultOptions.text_domain ) }</h2>
					<div className={ 'awesomemotive-settings-form-input-inner' }>
						{ Object.keys( emails ).map( ( value, c ) => {
							return (
								<EmailInputComponent
									key={ value }
									keyId={ parseInt( value ) + 1 }
									keyIndex={ value }
									value={ emails[ value ] ?? '' }
									saveEmail={ sendEmail }
									removeEmail={ removeEmail }
								/>
							);
						} ) }
						<div
							className={ 'awesomemotive-settings-add-inputs' }
							data-testid={ 'awesomemotive-settings-add-inputs' }
							onClick={ setNewInput }
						>
							<Icon icon={ plus } />
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};
