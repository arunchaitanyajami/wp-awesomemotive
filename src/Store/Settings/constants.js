import WpamEditorSettings from '../../deafultOptions';
export const STORE_NAME = 'awesomemotive/settings';
export const DEFAULT_STATE = {
	data: {
		num_rows: 5,
		human_date: false,
		emails: {},
		...WpamEditorSettings.settings,
	},
	status: 200,
	message: 'Data Obtained successfully',
	isInitialLoad: false,
	isDataUpdated: false,
};
