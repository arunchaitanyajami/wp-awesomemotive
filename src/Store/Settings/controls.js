import deafultOptions from '../../deafultOptions';

/**
 * Fetch data from source endpoint and update store.
 *
 * @param ajaxActionName
 * @param ajaxActionType
 * @param data
 * @returns {Promise<any>}
 * @constructor
 */
export function FETCH_FROM_API( { ajaxActionName, ajaxActionType, data } ) {
	const formData = new FormData();

	formData.append( 'action', ajaxActionName );
	formData.append( 'type', ajaxActionType );
	formData.append( 'wpam_nonce', deafultOptions.wp_nonce );
	if ( data ) {
		formData.append( 'data', data );
	}

	const params = new URLSearchParams( formData );

	return fetch( deafultOptions.ajaxUrl, {
		method: 'POST',
		credentials: 'same-origin',
		headers: {
			'Content-Type': 'application/x-www-form-urlencoded',
			'Cache-Control': 'no-cache',
		},
		body: params,
	} )
		.then( ( response ) => {
			return response.json();
		} )
		.then( ( response ) => {
			return response;
		} );
}
