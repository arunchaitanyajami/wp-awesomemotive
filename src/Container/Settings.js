import { compose } from '@wordpress/compose';
import { withDispatch, withSelect } from '@wordpress/data';

import Settings from '../Component/Settings';
import { STORE_NAME } from '../Store/Settings/constants';

export const mapSelectToProps = ( select ) => {
	const { getSettings } = select( STORE_NAME );

	return {
		settings: getSettings(),
	};
};

export const mapDispatchToProps = ( dispatch ) => {
	const { updateEmails, updateRows, updateDateReadable } =
		dispatch( STORE_NAME );

	return {
		updateEmails,
		updateRows,
		updateDateReadable,
	};
};

export default compose( [
	withSelect( mapSelectToProps ),
	withDispatch( mapDispatchToProps ),
] )( Settings );
