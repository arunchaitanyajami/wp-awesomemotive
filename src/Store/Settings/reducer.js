import * as constant from './constants';

const { DEFAULT_STATE } = constant;

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case 'SET_SETTINGS':
			return {
				...state,
				...action.settings,
			};
		case 'UPDATE_SETTINGS':
			return {
				...state,
				data: {
					...state.data,
					...{
						[ action.key ]: action.value,
					},
				},
				...action.settings,
			};
		case 'SET_DATA_LOAD_FLAG':
			return {
				...state,
				...{
					isInitialLoad: action.isInitialLoad,
				},
			};
		case 'SET_DATA_UPDATE_FLAG':
			return {
				...state,
				...{
					isDataUpdated: action.isDataUpdated,
				},
			};
	}

	return state;
};
