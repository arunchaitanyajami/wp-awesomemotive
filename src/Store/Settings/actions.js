import { FETCH_FROM_API } from './controls';

export function* updateOption( key, value ) {
	yield FETCH_FROM_API( {
		ajaxActionName: 'all_settings_endpoint',
		ajaxActionType: 'update',
		data: JSON.stringify( {
			[ key ]: value,
		} ),
	} );

	return {
		type: 'UPDATE_SETTINGS',
		key,
		value,
	};
}

export function deleteOption( key ) {
	return {
		type: 'DELETE_SETTINGS',
		key,
	};
}

export function setOptions( settings ) {
	return {
		type: 'SET_SETTINGS',
		settings,
	};
}

export function fetchFromAPI( ajaxActionName, ajaxActionType ) {
	return {
		type: 'FETCH_FROM_API',
		ajaxActionName,
		ajaxActionType,
	};
}

export function* updateEmails( emails ) {
	let emailsArray = Object.keys( emails ).map( function ( key ) {
		return emails[ key ];
	} );
	yield updateOption( 'emails', emailsArray );
}

export function* updateRows( rows ) {
	yield updateOption( 'num_rows', parseInt( rows ) );
}

export function* updateDateReadable( boolValue ) {
	yield updateOption( 'human_date', boolValue );
}
