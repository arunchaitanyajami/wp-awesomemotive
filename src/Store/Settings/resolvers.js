import { fetchFromAPI, setOptions } from './actions';

/**
 * On store select, fetch from endpoint and update the store and return store.
 *
 * @returns {Generator<{ajaxActionType, type: string, ajaxActionName}, {settings, type: string}, *>}
 */
export function* getSettings() {
	const settings = yield fetchFromAPI( 'all_settings_endpoint', 'get' );

	return setOptions({...settings, ...{isInitialLoad: true}});
}
