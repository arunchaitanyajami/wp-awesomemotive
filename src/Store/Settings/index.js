import { createReduxStore, register } from '@wordpress/data';

import * as constant from './constants';
import * as selectors from './selectors';
import * as actions from './actions';
import reducer from './reducer';
import * as resolvers from './resolvers';
import * as controls from './controls';

const settings = createReduxStore( constant.STORE_NAME, {
	reducer,
	actions,
	selectors,
	controls,
	resolvers,
} );

register( settings );
