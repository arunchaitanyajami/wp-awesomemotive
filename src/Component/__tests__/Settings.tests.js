import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Settings from '../Settings';

let props = {
	setValue: jest.fn(),
	setDate: jest.fn(),
	setEmails: jest.fn(),
	data: {
		emails: [ 'test@test.com' ],
		num_rows: 5,
		human_date: false,
	}
};

jest.mock( '@wordpress/data', () => ( {
	useSelect: jest.fn( () => ( {
		getSettings: jest.fn().mockReturnValue( {
			data: jest.fn().mockReturnValue( {
				emails: [ 'test@test.com' ],
				num_rows: 5,
				human_date: false,
			} ),
		} ),
	} ) ),
	withSelect: jest.fn(),
	combineReducers: jest.fn(),
	createReduxStore: jest.fn(),
	register: jest.fn(),
} ) );

describe( 'Settings Component', () => {
	let useEffect;
	let useState;
	beforeEach( () => {
		useEffect = jest.spyOn( React, 'useEffect' );
		useState = jest.spyOn( React, 'useState' );
	} );

	afterEach( () => {
		jest.clearAllMocks();
	} );

	const mockUseEffect = () => {
		useEffect.mockImplementationOnce( ( f ) => f() );
	};

	it( 'Should Render the default component', async () => {
		mockUseEffect();
		mockUseEffect();
		useState
			.mockReturnValueOnce( [ props.data.num_rows, jest.fn() ] )
			.mockReturnValueOnce( [ props.data.human_date, jest.fn() ] )
			.mockReturnValueOnce( [ props.data.emails, jest.fn() ] );
		const { container } = render( <Settings settings={props} /> );

		expect(
			parseInt(
				container.querySelector(
					'#awesomemotive-settings-form-input-no-of-rows'
				).value
			)
		).toBe( 5 );
		expect( container.querySelector( '#email_1' ).value ).toBe(
			props.data.emails[ 0 ]
		);
		expect(
			container.querySelector( '.components-form-toggle__input' )
		).toBeVisible();
		expect( container ).toMatchSnapshot();
	} );

	it( 'Should add multiple emails and check if node exists', async () => {
		props.data.emails = props.data.emails.concat( [
			'test2@test.com',
			'test3@test.com',
			'test4@test.com',
		] );

		mockUseEffect();
		mockUseEffect();
		useState
			.mockReturnValueOnce( [ props.data.num_rows, jest.fn() ] )
			.mockReturnValueOnce( [ props.data.human_date, jest.fn() ] )
			.mockReturnValueOnce( [ props.data.emails, jest.fn() ] );
		const { container } = render( <Settings settings={props} /> );

		expect( container.querySelector( '#email_1' ).value ).toBe(
			props.data.emails[ 0 ]
		);
		expect( container.querySelector( '#email_3' ).value ).toBe(
			props.data.emails[ 2 ]
		);

		expect( container ).toMatchSnapshot();
	} );
} );
