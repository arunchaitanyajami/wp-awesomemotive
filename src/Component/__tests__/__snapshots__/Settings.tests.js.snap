// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component Should Render the default component 1`] = `
<div>
  <div
    class="awesomemotive-settings-container"
  >
    <div
      class="awesomemotive-settings-form-inputs"
    >
      <div
        class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
        data-wp-c16t="true"
        data-wp-component="Flex"
      >
        <div
          class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
          data-wp-c16t="true"
          data-wp-component="FlexItem"
        >
          <label
            class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
            data-wp-c16t="true"
            data-wp-component="Text"
            for="awesomemotive-settings-form-input-no-of-rows"
          >
            No of rows
          </label>
        </div>
        <div
          class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
        >
          <input
            class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
            id="awesomemotive-settings-form-input-no-of-rows"
            type="number"
            value="5"
          />
          <div
            aria-hidden="true"
            class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
          />
        </div>
      </div>
    </div>
    <div
      class="awesomemotive-settings-form-inputs"
    >
      <div
        class="components-base-control components-toggle-control css-wdf2ti-Wrapper ej5x27r4"
      >
        <div
          class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
        >
          <span
            class="components-form-toggle"
          >
            <input
              aria-describedby="inspector-toggle-control-0__help"
              class="components-form-toggle__input"
              id="inspector-toggle-control-0"
              type="checkbox"
            />
            <span
              class="components-form-toggle__track"
            />
            <span
              class="components-form-toggle__thumb"
            />
          </span>
          <label
            class="components-toggle-control__label"
            for="inspector-toggle-control-0"
          >
            Human Readable Date
          </label>
        </div>
        <p
          class="components-base-control__help css-v0euhr-StyledHelp-deprecatedMarginHelp ej5x27r1"
          id="inspector-toggle-control-0__help"
        >
          Timestamp displayed
        </p>
      </div>
    </div>
    <div
      class="awesomemotive-settings-form-inputs"
    >
      <h2>
        Email List:
      </h2>
      <div
        class="awesomemotive-settings-form-input-inner"
      >
        <div
          class="awesomemotive-settings-form-input-repeater"
          id="awesomemotive-settings-form-input-repeater1"
        >
          <div
            class="awesomemotive-settings-form-input-control"
          >
            <div
              class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <label
                  class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
                  data-wp-c16t="true"
                  data-wp-component="Text"
                  for="email_1"
                >
                  Email 1
                </label>
              </div>
              <div
                class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
              >
                <input
                  class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
                  id="email_1"
                  name="email"
                  placeholder="Enter email address.."
                  required=""
                  type="email"
                  value="test@test.com"
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
                />
              </div>
            </div>
          </div>
          <div
            class="awesomemotive-settings-form-input-trash-icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 5h-5.7c0-1.3-1-2.3-2.3-2.3S9.7 3.7 9.7 5H4v2h1.5v.3l1.7 11.1c.1 1 1 1.7 2 1.7h5.7c1 0 1.8-.7 2-1.7l1.7-11.1V7H20V5zm-3.2 2l-1.7 11.1c0 .1-.1.2-.3.2H9.1c-.1 0-.3-.1-.3-.2L7.2 7h9.6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="awesomemotive-settings-add-inputs"
          data-testid="awesomemotive-settings-add-inputs"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 11.2h-5.2V6h-1.6v5.2H6v1.6h5.2V18h1.6v-5.2H18z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings Component Should add multiple emails and check if node exists 1`] = `
<div>
  <div
    class="awesomemotive-settings-container"
  >
    <div
      class="awesomemotive-settings-form-inputs"
    >
      <div
        class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
        data-wp-c16t="true"
        data-wp-component="Flex"
      >
        <div
          class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
          data-wp-c16t="true"
          data-wp-component="FlexItem"
        >
          <label
            class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
            data-wp-c16t="true"
            data-wp-component="Text"
            for="awesomemotive-settings-form-input-no-of-rows"
          >
            No of rows
          </label>
        </div>
        <div
          class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
        >
          <input
            class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
            id="awesomemotive-settings-form-input-no-of-rows"
            type="number"
            value="5"
          />
          <div
            aria-hidden="true"
            class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
          />
        </div>
      </div>
    </div>
    <div
      class="awesomemotive-settings-form-inputs"
    >
      <div
        class="components-base-control components-toggle-control css-wdf2ti-Wrapper ej5x27r4"
      >
        <div
          class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
        >
          <span
            class="components-form-toggle"
          >
            <input
              aria-describedby="inspector-toggle-control-1__help"
              class="components-form-toggle__input"
              id="inspector-toggle-control-1"
              type="checkbox"
            />
            <span
              class="components-form-toggle__track"
            />
            <span
              class="components-form-toggle__thumb"
            />
          </span>
          <label
            class="components-toggle-control__label"
            for="inspector-toggle-control-1"
          >
            Human Readable Date
          </label>
        </div>
        <p
          class="components-base-control__help css-v0euhr-StyledHelp-deprecatedMarginHelp ej5x27r1"
          id="inspector-toggle-control-1__help"
        >
          Timestamp displayed
        </p>
      </div>
    </div>
    <div
      class="awesomemotive-settings-form-inputs"
    >
      <h2>
        Email List:
      </h2>
      <div
        class="awesomemotive-settings-form-input-inner"
      >
        <div
          class="awesomemotive-settings-form-input-repeater"
          id="awesomemotive-settings-form-input-repeater1"
        >
          <div
            class="awesomemotive-settings-form-input-control"
          >
            <div
              class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <label
                  class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
                  data-wp-c16t="true"
                  data-wp-component="Text"
                  for="email_1"
                >
                  Email 1
                </label>
              </div>
              <div
                class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
              >
                <input
                  class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
                  id="email_1"
                  name="email"
                  placeholder="Enter email address.."
                  required=""
                  type="email"
                  value="test@test.com"
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
                />
              </div>
            </div>
          </div>
          <div
            class="awesomemotive-settings-form-input-trash-icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 5h-5.7c0-1.3-1-2.3-2.3-2.3S9.7 3.7 9.7 5H4v2h1.5v.3l1.7 11.1c.1 1 1 1.7 2 1.7h5.7c1 0 1.8-.7 2-1.7l1.7-11.1V7H20V5zm-3.2 2l-1.7 11.1c0 .1-.1.2-.3.2H9.1c-.1 0-.3-.1-.3-.2L7.2 7h9.6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="awesomemotive-settings-form-input-repeater"
          id="awesomemotive-settings-form-input-repeater2"
        >
          <div
            class="awesomemotive-settings-form-input-control"
          >
            <div
              class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <label
                  class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
                  data-wp-c16t="true"
                  data-wp-component="Text"
                  for="email_2"
                >
                  Email 2
                </label>
              </div>
              <div
                class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
              >
                <input
                  class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
                  id="email_2"
                  name="email"
                  placeholder="Enter email address.."
                  required=""
                  type="email"
                  value="test2@test.com"
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
                />
              </div>
            </div>
          </div>
          <div
            class="awesomemotive-settings-form-input-trash-icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 5h-5.7c0-1.3-1-2.3-2.3-2.3S9.7 3.7 9.7 5H4v2h1.5v.3l1.7 11.1c.1 1 1 1.7 2 1.7h5.7c1 0 1.8-.7 2-1.7l1.7-11.1V7H20V5zm-3.2 2l-1.7 11.1c0 .1-.1.2-.3.2H9.1c-.1 0-.3-.1-.3-.2L7.2 7h9.6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="awesomemotive-settings-form-input-repeater"
          id="awesomemotive-settings-form-input-repeater3"
        >
          <div
            class="awesomemotive-settings-form-input-control"
          >
            <div
              class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <label
                  class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
                  data-wp-c16t="true"
                  data-wp-component="Text"
                  for="email_3"
                >
                  Email 3
                </label>
              </div>
              <div
                class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
              >
                <input
                  class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
                  id="email_3"
                  name="email"
                  placeholder="Enter email address.."
                  required=""
                  type="email"
                  value="test3@test.com"
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
                />
              </div>
            </div>
          </div>
          <div
            class="awesomemotive-settings-form-input-trash-icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 5h-5.7c0-1.3-1-2.3-2.3-2.3S9.7 3.7 9.7 5H4v2h1.5v.3l1.7 11.1c.1 1 1 1.7 2 1.7h5.7c1 0 1.8-.7 2-1.7l1.7-11.1V7H20V5zm-3.2 2l-1.7 11.1c0 .1-.1.2-.3.2H9.1c-.1 0-.3-.1-.3-.2L7.2 7h9.6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="awesomemotive-settings-form-input-repeater"
          id="awesomemotive-settings-form-input-repeater4"
        >
          <div
            class="awesomemotive-settings-form-input-control"
          >
            <div
              class="components-flex components-input-control em5sgkm7 css-18gj5cw-View-Flex-sx-Base-sx-Items-ItemsColumn-Root-rootLabelPositionStyles em57xhy0"
              data-wp-c16t="true"
              data-wp-component="Flex"
            >
              <div
                class="components-flex-item em5sgkm3 css-1fmchc6-View-Item-sx-Base-LabelWrapper em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <label
                  class="components-truncate components-text components-input-control__label em5sgkm4 css-1790joj-View-Text-sx-Base-BaseLabel-labelMargin em57xhy0"
                  data-wp-c16t="true"
                  data-wp-component="Text"
                  for="email_4"
                >
                  Email 4
                </label>
              </div>
              <div
                class="components-input-control__container css-1t6cb2b-Container-containerDisabledStyles-containerWidthStyles em5sgkm6"
              >
                <input
                  class="components-input-control__input css-gzm6eu-Input-dragStyles-fontSizeStyles-sizeStyles em5sgkm5"
                  id="email_4"
                  name="email"
                  placeholder="Enter email address.."
                  required=""
                  type="email"
                  value="test4@test.com"
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-1gb9p1j-BackdropUI-backdropFocusedStyles em5sgkm2"
                />
              </div>
            </div>
          </div>
          <div
            class="awesomemotive-settings-form-input-trash-icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 5h-5.7c0-1.3-1-2.3-2.3-2.3S9.7 3.7 9.7 5H4v2h1.5v.3l1.7 11.1c.1 1 1 1.7 2 1.7h5.7c1 0 1.8-.7 2-1.7l1.7-11.1V7H20V5zm-3.2 2l-1.7 11.1c0 .1-.1.2-.3.2H9.1c-.1 0-.3-.1-.3-.2L7.2 7h9.6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="awesomemotive-settings-add-inputs"
          data-testid="awesomemotive-settings-add-inputs"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 11.2h-5.2V6h-1.6v5.2H6v1.6h5.2V18h1.6v-5.2H18z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
